### ISSUE TITLE
Capture the singleton type of the term of the record or enum and pass it through the accessor builder

### ISSUE THREAD

**Issue by sviezypan** [zio/zio-schema#242]
I need a singleton type of the term of the record or enum passed through AccessorBuilder, so something like:
```
trait AccessorBuilder {
  type Lens[F, S, A]
  type Prism[F, S, A]
  type Traversal[F, S, A]

  def makeLens[F, S, A](product: Schema.Record[S], term: Schema.Field[A]): Lens[F, S, A]

  def makePrism[F, S, A](sum: Schema.Enum[S], term: Schema.Case[A, S]): Prism[F, S, A]

  def makeTraversal[F, S, A](collection: Schema.Collection[S, A], element: Schema[A]): Traversal[F, S, A]
}
```

This need to modify also Accessors in Schema to 

```
type Accessors[Lens[_, _, _], Prism[_, _, _], Traversal[_, _, _]]
```

**Comment by tusharmath** [comment]
If `S` is the whole object and `A` is a part of it, then what does `F` represent here?

**Comment by sviezypan** [comment]
`F` would be a singleton type equal to the name of `A`, 

Maybe not useful in lot of places but lets say that we have 
`case class Person(name: String, age: Int)` 

and we need a type that represents tuple of its field names
`type FieldNames = ("name", "age")`

Then we can create an accessor builder like:
```
val identities = new AccessorBuilder {
    override type Lens[F, S, A] = F

    override type Prism[F, S, A] = Unit

    override type Traversal[S, A] = Unit

    def makeLens[F, S, A](product: Schema.Record[S], term: Schema.Field[A]): Lens[F, S, A] =
        term.label.asInstanceOf[F]

    def makePrism[F, S, A](sum: Schema.Enum[S], term: Schema.Case[A, S]): Prism[F, S, A] = ()

    def makeTraversal[S, A](collection: Schema.Collection[S, A], element: Schema[A]): Traversal[S, A] = ()
  }
  ```
And when we have personSchema we can call 
  ```
  type FieldNames = 
        personSchema.Accessors[
          identities.Lens,
          identities.Prism,
          identities.Traversal
        ]
 ```

**Comment by sviezypan** [comment]
Implemented in #246
